### PR TITLE
Validate image references before registry requests

### LINKED ISSUE #314: Validate image references before registry requests
## Problem

Image references are parsed permissively, allowing malformed repositories, tags, digests, or registry components to proceed until a later registry request fails. This delays feedback and can produce diagnostics unrelated to the actual input error.

## Proposed improvement

Validate image-reference syntax at the CLI boundary before network access. Report which component is invalid and show the accepted forms, while preserving supported Docker Hub shorthand such as `ubuntu` and `owner/image:tag`.

Add tests for empty components, malformed digest syntax, invalid tags, registry ports, and valid shorthand references.

> This issue was generated by GitHub Copilot from an AI-assisted CLI API and UX review.

### PR DESCRIPTION
**Pull request by mthalman**
Malformed image references currently reach registry client creation, producing avoidable network requests and less actionable failures. This change validates references at the CLI boundary so invalid registry, repository, tag, and digest components are reported immediately.

## Summary

- Add component-aware image reference parsing with Docker-compatible registry, repository, tag, digest, port, IPv4, and IPv6 validation.
- Reuse a shared System.CommandLine validator across image, manifest, referrer, comparison, and tag-list arguments.
- Preserve Docker Hub shorthand and normalization, while requiring `tag list` to receive a repository without a tag or digest.
- Add parser boundaries, CLI diagnostics, command wiring, and no-registry-client regression coverage.

## Testing

- `dotnet test src\Valleysoft.Dredge.sln --no-restore -v normal -c Release`

Fixes: #314

### CODE REVIEW

File: src/Valleysoft.Dredge/ImageName.cs [hunk 1]
```diff
@@ -17,33 +36,107 @@ public ImageName(string? registry, string repo, string? tag, string? digest)
...
         int separatorIndex = imageName.IndexOf('/');
         if (separatorIndex >= 0)
         {
             string firstSegment = imageName[..separatorIndex];
-            if (firstSegment.Contains('.') || firstSegment.Contains(':'))
+            if (firstSegment.Contains('.') ||
+                firstSegment.Contains(':') ||
+                firstSegment.Equals("localhost", StringComparison.OrdinalIgnoreCase) ||
+                firstSegment.StartsWith('[') ||
+                firstSegment.Any(char.IsAsciiLetterUpper))
```
[the remark this answers is not in the document]
Comment (mthalman) [reply src/Valleysoft.Dredge/ImageName.cs:95]: I’m retaining this behavior because it matches Docker’s canonical `distribution/reference` implementation: `splitDockerDomain` explicitly treats a first component containing uppercase characters as the domain since repository path components must be lowercase. Thus `Org/image` is normalized as host `Org` plus repository `image`, while lowercase `org/image` remains Docker Hub namespace shorthand.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/Valleysoft.Dredge/Commands/ImageReferenceArgument.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            Description = $"{subject} reference {Syntax}"
+        };
+        argument.Validators.Add(result =>
+        {
+            string? value = result.GetValueOrDefault<string>();
+            if (!ImageName.TryParse(value, out _, out string? error))
+            {
+                result.AddError(
+                    $"{error} Expected <image>, <image>:<tag>, or <image>@<digest>.");
+            }
```
[the remark this answers is not in the document]
Comment (mthalman) [reply src/Valleysoft.Dredge/Commands/ImageReferenceArgument.cs:22]: Updated the validator to derive accepted forms from the actual argument name, so comparison errors now say `<base>` or `<target>` while ordinary image arguments continue to say `<image>`. Added regression coverage for both comparison arguments.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._
